### ISSUE TITLE
Track loops indefinitely

### ISSUE THREAD

**Issue by stevekuznetsov** [pithos/pithos#409]
As seen in the currently playing song, the amount of time it's been playing for is longer than the reported length of the track:

![pithos-timing](https://cloud.githubusercontent.com/assets/7328370/22296829/6233f376-e2e9-11e6-9051-55cdc1b48cd4.png)

I'm running from a source install at:
```
d77de16f20d1fb368cefe7db6cd44159a88cc0fe fix not reconnecting on PandoraAuthTokenInvalid error.
```

Nothing interesting in the journal logs... I think the song is just playing twice in a row? I'm not getting debug output so I can't share that.

**Comment by stevekuznetsov** [comment]
This track is just looping over and over.

**Comment by stevekuznetsov** [comment]
Skipping to new tracks after that did not continue to display this bug.

**Comment by JasonLG1979** [comment]
That is weird. We use the eos (end of stream) signal to trigger song changes. Which is pretty much textbook as far as Gstreamer useage. I don't know why the song would have looped over again? I've had Pandora give me playlists with duplicate songs before but I've never seen this. Has it only happened once?

**Comment by stevekuznetsov** [comment]
Yeah only happened the once today when I upgraded to the latest `master`.

**Comment by JasonLG1979** [comment]
Are you familiar with the song? Did you hear it loop?

**Comment by stevekuznetsov** [comment]
Yes, I'm familiar with it. It started looping the third time and I skipped forward. It played through once, then started again (when I made the screenshot) then once the second loop was over, started again.

**Comment by stevekuznetsov** [comment]
Is there any way to turn on debugging output from `pithos` at run-time? Or a rotating log so that I could always run with it on, but not have it take up more than a couple MB of space?

**Comment by JasonLG1979** [comment]
That either got to be a Pandora or Gstreamer issue.

**Comment by JasonLG1979** [comment]
Look in your logs. I believe it uses systemd logging

**Comment by stevekuznetsov** [comment]
As I said, the output from `journalctl` is not super interesting. 

```
# journalctl -t io.github.Pithos
...
Jan 25 10:09:02 io.github.Pithos[3980]: EOS
Jan 25 10:09:02 io.github.Pithos[3980]: Starting song: index = 13
Jan 25 10:09:02 io.github.Pithos[3980]: Using audio quality highQuality: 192 mp3
Jan 25 10:09:02 io.github.Pithos[3980]: Using audio quality highQuality: 192 mp3
Jan 25 10:09:02 io.github.Pithos[3980]: Not an Ad..
Jan 25 10:09:02 io.github.Pithos[3980]: Using audio quality highQuality: 192 mp3
Jan 25 10:34:34 io.github.Pithos[3980]: pandora: addFeedback
Jan 25 10:47:20 io.github.Pithos[3980]: Starting song: index = 14
Jan 25 10:47:20 io.github.Pithos[3980]: Using audio quality highQuality: 192 mp3
Jan 25 10:47:20 io.github.Pithos[3980]: Using audio quality highQuality: 192 mp3
Jan 25 10:47:21 io.github.Pithos[3980]: Not an Ad..
Jan 25 10:47:21 io.github.Pithos[3980]: Using audio quality highQuality: 192 mp3
Jan 25 10:53:43 io.github.Pithos[3980]: EOS
...
```

You can see the song begin at 10:09, after two loops of 18 minutes (36 minutes) I skipped forward at 10:47. Looks like no EOS occurred. The next song ended as expected.

**Comment by JasonLG1979** [comment]
That is somewhat helpful but still weird. I'll start a station with that song as the seed and see if can't duplicate the issue.

**Comment by JasonLG1979** [comment]
I can duplicate your issue with another song by the same artist from the same album. 
![screenshot from 2017-01-25 16-53-11](https://cloud.githubusercontent.com/assets/6667703/22312794/137fbbbc-e31f-11e6-8831-639f7c18ed08.png)

If I had to guess it's Pandora's problem. The next step is to actually download the mp3 and see how long it actually is.

**Comment by stevekuznetsov** [comment]
What a weird bug!

**Comment by JasonLG1979** [comment]
As soon as my test account gets unblocked(I requested too many playlists trying to get those songs) I'll do some digging. But IDK? Either Gstreamer is being weird or whoever submitted it to Pandora looped the song a few times when they converted it to MP3? So far it only seems to effect those 2 songs so maybe the whole album is borked?

**Comment by JasonLG1979** [comment]
Well, I've got a couple shorter songs from that album and they play as expected.

**Comment by stevekuznetsov** [comment]
Interesting. Maybe we should just submit the issue to Pandora, then. I can't say I remember definitively that I've ever listened to that song and not had it repeat. I've definitely listened to some on the album, though, and not had any issues.

**Comment by JasonLG1979** [comment]
Well I downloaded an offending song and played it locally locally in several players and it plays as it should I also played the url in Exaile and a little test app I wrote and they both play the stream as expected(and get an EOS) for some reason Pithos is missing the EOS. Where did you update from commit wise?

**Comment by JasonLG1979** [comment]
> Interesting. Maybe we should just submit the issue to Pandora

Not a good idea. We're using their private API. We exist somewhat in a gray area. The less noise we make the less likely they are to change their API thus breaking Pithos.

**Comment by JasonLG1979** [comment]
@stevekuznetsov I found the problem. In a previous commit I set the network timeout in httpsoupsrc to 10min from the default 10 sec. My logic was that it would give the user enough time to reconnect if their connection was dropped, but it seems to have the unintended side effect of making Pithos miss the EOS signal on longer songs. I'm not sure why but reverting that change seems to fix the problem.
